### DISCUSSION TITLE
Is there a highlight for builtin matchit?

### DISCUSSION THREAD

**Discussion by ndavd** [neovim/neovim#29205] [Q&A] [answered]
Hi, I found that neovim's builtin `matchit` works very well and I don't need an external plugin to improve it. But upon removing that plugin my highlighting of matched words also got removed. I searched `matchit` docs but found no highlight groups. Is there one? If not, why isn't this a builtin feature?

**Comment by clason** [comment]
The highlight doesn't come from `matchit` but from `matchparen` and is -- wait for it -- `MatchParen`.

**Reply by ndavd** [reply]
Hmm interestingly I have `MatchParen guibg=#303030` and yet I see no highlighting 🤔

**Reply by clason** [reply]
See if something else overrides it, and if `termguicolors` are set.

**Reply by ndavd** [reply]
`termguicolors` are set and `:hi MatchParen` returns me the highlight above

**Reply by clason** [reply]
Try with `--clean` -- something else must be interfering then.

**Reply by ndavd** [reply]
For me `MatchParen` highlight is only working on `{}`, `[]`, etc. Not on actual words, like `if end`, `function end`, etc.

EDIT: Upon reading the docs it says it highlights the character under the cursor, not the word. So I'm guessing that the functionality I mentioned is indeed missing.

**Reply by zeertzjq** [reply]
Those are not supported by matchparen.

**Reply by ndavd** [reply]
When I do `%` on a lua `function` `end` it hops, even if I disabled Treesitter and LSP. So it seems builtin

**Reply by clason** [reply, marked as answer]
> When I do % on a lua function end it hops, even if I disabled Treesitter and LSP. So it seems builtin

Yes, that is `matchit` (which is bundled and therefore loaded with `--clean`), which has basic "generic syntax" (read: C) support builtin. But it doesn't highlight anything; the only highlights you get is from matching _delimiters_ via `matchparen`.

**Reply by ndavd** [reply]
Oh that's unfortunate, it would be great if there was `MatchIt` highlight or something (could perhaps be linked to `MatchParen` by default). Do you think it's worth it to open a feature request?

**Reply by clason** [reply]
Not here; you can open an issue at the matchit repo https://github.com/chrisbra/matchit. (~~But check if the standalone plugin already has that; if so, you can ask chrisbra to update the bundled version in Vim.~~ It does not.)

**Reply by ndavd** [reply]
Thank you!

**Reply by ndavd** [reply]
https://github.com/chrisbra/matchit/issues/45
